akita/src/internal/crud.ts

```diff
@@ -92,15 +104,25 @@ export class CRUD {
         newEntity = new oldEntity.constructor(merged);
       }
 
-      updatedEntities[id] = newEntity;
+      updatedEntities[idToUpdate] = newEntity;
+    }
+
+    let updatedIds = state.ids;
+    let stateEntities = state.entities;
+    if (isUpdatingIdKey) {
+      const id = ids[0];
+      const { [id]: any, ...rest } = state.entities;
+      stateEntities = rest;
+      updatedIds = [...state.ids.filter(current => current !== id), idToUpdate];
```

Comment:
Notice that you are changing the order here. You should keep the entities order otherwise the view will change. Just replace the old id with the new one.

Reply:
good catch!
Can't "just replace" the old id with new id because in dev mode the ids are frozen. I'll use `.map` to do that.